src/main/java/keycontacts/model/lesson/Lesson.java

```diff
@@ -29,6 +29,52 @@ public Lesson(Time startTime, Time endTime) {
         this.endTime = endTime;
     }
 
+    /**
+     * Returns true if any 2 lessons clash.
+     */
+    public static boolean lessonsClash(Lesson lesson1, Lesson lesson2) {
```

Comment:
Firstly, this method does not follow naming convention for boolean methods.

Secondly, I think the implementation can be made clearer by
Turn it into a class method.
Lesson has two abstract methods `isClash(RegularLesson)` and `isClash(MakeupLesson)` which both RegularLesson and MakeupLesson can override.